akka-actor/src/main/scala/akka/routing/RoundRobin.scala

```diff
@@ -23,11 +21,16 @@ object RoundRobinRoutingLogic {
  */
 @SerialVersionUID(1L)
 final class RoundRobinRoutingLogic extends RoutingLogic {
-  val next = new AtomicLong(0)
+  val next = new AtomicInteger
 
   override def select(message: Any, routees: immutable.IndexedSeq[Routee]): Routee =
     if (routees.isEmpty) NoRoutee
-    else routees((next.getAndIncrement % routees.size).asInstanceOf[Int])
+    else {
+      val nextSeq = next.getAndIncrement
+      // If nextSeq is Integer.MAX_VALUE + 1 then nextSeq is now Integer.MIN_VALUE resulting in a negative modulus operation.
+      // Math.abs would at inverted the order breaking the continuity of the sequence: N, N - 1, ..., 0, N, ....
+      routees(if (nextSeq < 0) routees.size + (nextSeq % routees.size) else nextSeq % routees.size)
```

Comment:
Well, there are 3 options here each with is draw back:
* AtomicInteger: Occasional jump but no need to cast from int to long.
* AtomicLong: Occasional jump unlikely to happen, always has to cast from long to int.
* AtomicInteger with CAS: The most correct solution but way slower.